src/app/core/market/order-status-notifier/order-status-notifier.service.ts

```diff
@@ -1,96 +1,126 @@
 import { Injectable, OnDestroy } from '@angular/core';
 import { Log } from 'ng2-logger';
 import { Observable } from 'rxjs/Observable';
+import * as _ from 'lodash';
 
 import { ListingService } from 'app/core/market/api/listing/listing.service';
 import { NotificationService } from 'app/core/notification/notification.service';
 import { MarketStateService } from 'app/core/market/market-state/market-state.service';
+import { ProfileService } from 'app/core/market/api/profile/profile.service';
+import { BidCollection } from 'app/core/market/api/bid/bidCollection.model';
+import { Bid } from 'app/core/market/api/bid/bid.model';
 
 @Injectable()
 export class OrderStatusNotifierService implements OnDestroy {
 
   log: any = Log.create('order-status-notifier.service id:' + Math.floor((Math.random() * 1000) + 1));
+
   private destroyed: boolean = false;
+  profile: any = {};
   private actionStatus: Array<any> =  [
             'Accept bid',
             'Mark as delivered',
             'Make payment',
             'Mark as shipped',
             'Order rejected'
           ];
-  public oldOrders: Array<any>;
-
+  public oldOrders: Bid[];
+  bids: BidCollection;
   constructor(
     private listingService: ListingService,
     private _marketState: MarketStateService,
-    private _notification: NotificationService
+    private _notification: NotificationService,
+    private profileService: ProfileService
   ) {
     this.log.d('order status notifier service running!');
+    this.loadOrders();
+  }
+
+  loadOrders() {
+    // @TODO need to replace with marketplace command so this should probably gone :)
+    this._marketState.observe('bid')
+      .takeWhile(() => !this.destroyed)
+      .map(o => new BidCollection(o, this.profile.address))
+      .subscribe(bids => {
+        this.bids = bids;
+        if (bids.address) {
+          this.checkForNewStatus(bids.orders);
+        }
+      })
+    this.loadProfile();
   }
 
+
   // TODO: trigger by ZMQ in the future
-  checkForNewStatus(orders: any): void {
+  checkForNewStatus(orders: Bid[]): void {
     this.log.d('new orders', orders);
     // if no orders: stop
-    if (orders.orders.length === 0) {
+    if (orders.length === 0) {
       return;
     }
-
-    if (!this.oldOrders || this.oldOrders.length === 0) {
-      this.oldOrders = orders.bid.orders;
-    } else {
-      this.checkOrders(orders.bid.orders);
-      this.oldOrders = orders.bid.orders;
+    orders.reverse();
```

Comment:
Move all the reverse()'ing into the bid collection model

Reply:
yup :)